src/components/Choropleth.vue

```diff
@@ -55,50 +53,84 @@ const backgroundLayer = {
 
 const map = ref<typeof LMap | null>(null);
 const featureRefs = ref<(typeof LGeoJson)[]>([]);
+const bounds: Ref<LatLngBounds | null> = ref(null);
 
 const router = useRouter();
-const { selectedFeatures, selectedIndicators, loading, selectedIndicator, selectedCountryId, appConfig } =
-    storeToRefs(useAppStore());
+const {
+    selectedFeatures,
+    selectedIndicators,
+    loading,
+    selectedIndicator,
+    selectedCountryId,
+    appConfig,
+    countryBoundingBoxes,
+    waitingForMapBounds
+} = storeToRefs(useAppStore());
+
+const showSpinner = computed(() => waitingForMapBounds.value || loading.value);
+useLoadingSpinner(map, showSpinner);
 
-useLoadingSpinner(map, loading);
 const { getColour } = useColourScale(selectedIndicators);
 
-const getFeatureId = (feature: Feature) => feature.properties![FEATURE_ID_PROP];
-const getFeatureName = (feature: Feature) => feature.properties![FEATURE_NAME_PROP];
+const featureProperties = computed(() => appConfig.value?.geoJsonFeatureProperties);
+
+const featureInSelectedCountry = (feature: Feature, selectedCountry) =>
+    feature.properties[featureProperties.value.country] === selectedCountry;
+const getFeatureId = (feature: Feature) =>
+    featureInSelectedCountry(feature, selectedCountryId.value)
+        ? feature.properties![featureProperties.value.idAdm2]
+        : feature.properties![featureProperties.value.idAdm1];
+const getFeatureName = (feature: Feature) =>
+    featureInSelectedCountry(feature, selectedCountryId.value)
+        ? feature.properties![featureProperties.value.nameAdm2]
+        : feature.properties![featureProperties.value.nameAdm1];
 
 const getColourForFeature = (feature, indicator) => {
-    const featureIndicators = selectedIndicators.value[getFeatureId(feature)];
+    const featureId = getFeatureId(feature);
+    const featureIndicators = selectedIndicators.value[featureId];
     return getColour(indicator, featureIndicators);
 };
 
 const featuresWithColours = computed(() => {
     const selectedInd = selectedIndicator.value;
-    if (!selectedInd) return [];
-    return selectedFeatures.value.map((feature) => {
-        return {
-            feature,
-            colour: getColourForFeature(feature, selectedInd)
-        };
-    });
+    if (!selectedInd) return {};
+    const selectedCountry = selectedCountryId.value;
+    return Object.fromEntries(
+        selectedFeatures.value.map((feature) => {
+            return [
+                getFeatureId(feature),
+                {
+                    feature,
+                    colour: getColourForFeature(feature, selectedInd, selectedCountry)
```

Comment:
also dont think you need to pass in selectedIndicator, you can just use it in the function but that dioesnt matter too much
```suggestion
                    colour: getColourForFeature(feature, selectedInd)
```

Reply:
Oh yeah, not sure what happened there! 😆